lib/types/zpool.nix

```diff
@@ -50,6 +53,55 @@
       });
       description = "List of datasets to define";
     };
+    topology = lib.mkOption {
+      type =
+        let
+          vdev = lib.types.submodule ({ name, ... }: {
+            options = {
+              mode = lib.mkOption {
+                type = lib.types.enum modeOptions;
+                default = "";
+                description = "mode of the zfs vdev";
+              };
+              members = lib.mkOption {
+                type = lib.types.listOf lib.types.str;
+                description = "Members of the vdev";
+              };
+            };
+          });
+          parent = config;
+        in
+        lib.types.nullOr
+          (lib.types.submodule
+            ({ config, name, ... }: {
+              options = {
+                type = lib.mkOption {
+                  type = lib.types.enum [ "zfs_topology" ];
+                  default = "zfs_topology";
+                  internal = true;
+                  description = "Type";
+                };
+                # zfs disk types
+                vdev = lib.mkOption {
+                  type = lib.types.listOf vdev;
+                  default = [ ];
+                  description = "A list of storage vdevs";
+                };
+                special = lib.mkOption {
+                  type = lib.types.nullOr vdev;
+                  default = null;
+                  description = "A list of devices for the special vdev";
```

Comment:
Can we link to some upstream documentation what special means? It's new to me.

Reply:
Expanded description with relevant headings in https://openzfs.github.io/openzfs-docs/man/master/7/zpoolconcepts.7.html